### PR TITLE
Remove the deprecated `pl.callbacks.ProgressBar`

### PR DESCRIPTION
**Pull request by rschireman**
## What does this PR do?

Fixes part of #12521 by removing deprecated `pytorch lightning.callbacks.progress.progress bar`

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: tests/deprecated_api/test_remove_1-7.py [hunk 1]
```diff
@@ -350,7 +348,6 @@ def test_v1_7_0_deprecated_slurm_job_id():
         trainer.slurm_job_id
 
 
-@RunIf(min_gpus=1)
```
Comment (carmocca) [inline tests/deprecated_api/test_remove_1-7.py:353]: Did you remove this by accident?

File: CHANGELOG.md [hunk 2]
```diff
@@ -75,7 +75,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed the deprecated `terminate_on_nan` argument from the `Trainer` constructor ([#12553](https://github.com/PyTorchLightning/pytorch-lightning/pull/12553))
 
 
--
+- Remove deprecated `pytorch lightning.callbacks.progress.progress bar`
```
Comment (carmocca) [inline CHANGELOG.md:78]:
```suggestion
- Remove the deprecated `pytorch_lightning.callbacks.ProgressBar` ([#12658](https://github.com/PyTorchLightning/pytorch-lightning/pull/12658))
```

**Review by carmocca — APPROVED** [review]

File: tests/deprecated_api/test_remove_1-7.py (hunk 1 above)
Comment (rschireman) [reply tests/deprecated_api/test_remove_1-7.py:353]: yes I removed it, I'll add it back

File: CHANGELOG.md (hunk 2 above)
Comment (awaelchli) [inline CHANGELOG.md:78]:
```suggestion
- Remove deprecated `pytorch_lightning.callbacks.progress.progress_bar`
```
The underscore is a bridge that connects the soul with the mind.

File: pytorch_lightning/callbacks/__init__.py [hunk 3]
```diff
@@ -45,7 +45,6 @@
     "ModelPruning",
     "ModelSummary",
     "BasePredictionWriter",
-    "ProgressBar",
```
Comment (awaelchli) [inline pytorch_lightning/callbacks/__init__.py:48]: I bet the next idea someone will have is to rename ProgressBarBase to ProgressBar 🤣

**Review by awaelchli — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -75,7 +75,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed the deprecated `terminate_on_nan` argument from the `Trainer` constructor ([#12553](https://github.com/PyTorchLightning/pytorch-lightning/pull/12553))
 
 
--
+- Remove deprecated `pytorch_lightning.callbacks.progress.progress_bar`
```
Comment (Borda) [inline CHANGELOG.md:78]:
```suggestion
- Remove deprecated `pytorch_lightning.callbacks.progress.progress_bar` ([#12658](https://github.com/PyTorchLightning/pytorch-lightning/pull/12658))
```

**Review by Borda — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -84,15 +84,18 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Removed the deprecated `terminate_on_nan` argument from the `Trainer` constructor ([#12553](https://github.com/PyTorchLightning/pytorch-lightning/pull/12553))
 
 
+- Remove deprecated `pytorch_lightning.callbacks.progress.progress_bar` ([#12658](https://github.com/PyTorchLightning/pytorch-lightning/pull/12658))
```
Comment (awaelchli) [inline CHANGELOG.md:87]:
```suggestion
- Remove deprecated `pytorch_lightning.callbacks.progress.progress` ([#12658](https://github.com/PyTorchLightning/pytorch-lightning/pull/12658))
```

**Review by carmocca — APPROVED** [review]

**Pull request merged**